### ISSUE TITLE
Get hit by 'Invalid Token := )'

### ISSUE THREAD

**Issue by YoEight** [haskell/haskell-mode#875]
Hi,
I have an issue with an invalid parsing tree again that prevents me from editing my buffer. Unfortunately, I unable to know where it comes from as the source file is quite big and `ghc` isn't complaining. I'm reiterating my previous suggestion. Is it possible to disable such feature, I mean being able to tune it through a setting? Personally, I don't ask that much from a language mode.

Thanks for your time.

**Comment by gracjan** [comment]
Can you copy and paste the declaration or function where you have this
token?

**Comment by YoEight** [comment]
Unfortunately, it's a proprietary code. I'll try to figure out what could cause this and post a contrived version. It might also be helpful to add the line and the column where 'Invalid Token' happens in the buffer.

**Comment by gracjan** [comment]
We need to work by example. If you cannot provide the code we will need to close this issue.

**Comment by YoEight** [comment]
Sure, I didn't find a contrived example yet but I got the same error in a different context:

```haskell
-- (in a where binding)
create :: forall a. Operation 'Init (ReadResult 'RegularStream ReadEvent) -> Input 'Init StreamMedataResult a -> a
```

**Comment by YoEight** [comment]
From experience, it seems it has to do either on `DataKinds`-like expression or `TemplateHaskell` splice.

**Comment by gracjan** [comment]
Fixed by #903.

**Comment by YoEight** [comment]
Thank you !
